src/index.js

```diff
@@ -1,160 +1,333 @@
 'use strict'
 
-const EE = require('events').EventEmitter
-const util = require('util')
+const EventEmitter = require('events')
 const TimeCache = require('time-cache')
+const values = require('lodash.values')
+const pull = require('pull-stream')
+const lp = require('pull-length-prefixed')
+const Pushable = require('pull-pushable')
+
+const Peer = require('./peer')
 const utils = require('./utils')
 const pb = require('./message')
 const config = require('./config')
+
 const log = config.log
-const _intersection = require('lodash.intersection')
-const dialOnFloodSub = require('./dial-floodsub.js')
-const mountFloodSub = require('./mount-floodsub.js')
-const _values = require('lodash.values')
+const multicodec = config.multicodec
+const ensureArray = utils.ensureArray
 
-module.exports = PubSubGossip
+/**
+ * PubSubGossip, also known as pubsub-flood or just dumbsub,
+ * this implementation of pubsub focused on delivering an API
+ * for Publish/Subscribe, but with no CastTree Forming
+ * (it just floods the network).
+ */
+class PubSubGossip extends EventEmitter {
+  /**
+   * @param {Object} libp2p
+   * @param {Object} dagService
+   * @returns {PubSubGossip}
+   */
+  constructor (libp2p, dagService) {
+    super()
 
-util.inherits(PubSubGossip, EE)
+    this.dag = dagService
```

Comment:
Want to move this to use `IPLD Resolver` instead? `DAGService` is now part of the past. Should not require any API changes, just rename (and change in the test)